### ISSUE TITLE
RATE_LIMITED on "Discussing changesets"

### ISSUE THREAD

**Issue by yrtimiD** [Zaczero/osm-revert#32]
After reverting around 50 changesets within a few minutes with "Auto-discussion" enabled, some discussion comment adding fails with RATE_LIMITED.

**Comment by Zaczero** [comment]
Hey! How would you want such situations to be handled? Currently it just skips over the failed discussions.

**Comment by yrtimiD** [comment]
Right, challenging situation. First thing I thought about is at least allowing manual retry action. User might need to wait some time before using the retry option. Also, if api response has a clue about lockout period - will be very nice to show it.

**Comment by Zaczero** [comment]
With how osm-revert is designed, a manual retry action is not trivial to do. Ideally, osm-revert would parse OSM response headers to understand current rate limits and adjust its operation. However, I couldn't find any public documentation on this. The good news is that OpenStreetMap-NG [already supports](https://github.com/Zaczero/openstreetmap-ng/blob/main/app/middlewares/rate_limit_middleware.py) rate limit headers, so at some point, this issue will be resolved.

**Comment by yrtimiD** [comment]
I see. Maybe at least we can linkify changets numbers in the log to make manual discussion adding slightly less painful?

**Comment by Zaczero** [comment]
Okay sounds really useful! I have created an issue but I cannot give any ETA on the completion date.
